semgrep/semgrep
# fix: surface error messages in ls e2e tests

Previously we were doing an Lwt.join that would wait for all promises to resolve before it resolved. This was done on a promise with the test, and a promise with the server. So if the server errored out, this wouldn't bubble up, and instead would be wrapped into the promise.

So what would happen is the test would never finish because the server errord out, but then since the test never resolved, the join didn't either. This meant the error would essentially be thrown away.

Now what we do is run the server in Lwt.async which automatically reports the failure as a test failure


## Test plan
* Change a function in the server code to "failwith 'message'"
* `make test`
* See the error message
* CI tests pass still

previously the test would timeout and you wouldn't see this
